actions/notify-slack/get-info.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+function get-info() {
+  set -eo pipefail
+
+  if [ -z "${EMOJI:-}" ]; then
+
+    # MESSAGE="A $TYPE for $APPLICATION:$VERSION to $ENVIRONMENT resulted in $STATUS"
+    case "${STATUS:-}" in
+    success)
+      EMOJI=large_green_circle
+      ;;
+    failure)
+      EMOJI=red_circle
+      ;;
+    started)
+      EMOJI=large_blue_circle
+      # MESSAGE="A $TYPE has been started for $APPLICATION:$VERSION to $ENVIRONMENT"
+      ;;
+    *)
+      EMOJI=white_circle
+      ;;
+    esac
+
+  fi
+  echo emoji="$EMOJI" >>"$GITHUB_OUTPUT"
+  TIMESTAMP="$(date +%s)"
+  MESSAGE="A ${TYPE:-}for ${APPLICATION:-}:${VERSION:-} to ${ENVIRONMENT:-} resulted in ${STATUS:-}"
+
+  if [ "${STATUS:-}" = "success" ]; then
+    echo "status=success" >>"$GITHUB_OUTPUT"
+    echo "message=$MESSAGE" >>"$GITHUB_OUTPUT"
+  else
+    echo "status=failure" >>"$GITHUB_OUTPUT"
+    echo "message=$MESSAGE" >>"$GITHUB_OUTPUT"
+  fi
```

Comment:
There are a few things wrong with this:

1. The if statement itself is redundant.  We are not changing anything between the if and the else.

```bash
    echo "status=${STATUS:-failure}" >>"$GITHUB_OUTPUT"
    echo "message=$MESSAGE" >>"$GITHUB_OUTPUT"
```
 would be the simplified version of this; however, this is still not what we want.  We want to change the message slightly if the status is started.

2. This only allows two types of status: success and failure.  If STATUS is not success you are setting it to failure.

3. We do not need to output the status in the first place.  We can simply pass ${{ inputs.status }} in its place within the action.  We are passing in STATUS to determine a default EMOJI and MESSAGE.

4. We talked about letting the user of this action pass in their own message.  We should not be setting MESSAGE if it is already set.  Please wrap the entirety of working with the MESSAGE in an if statement if MESSAGE is empty.